website/pages/docs/acl/auth-methods/oidc.mdx

```diff
@@ -0,0 +1,214 @@
+both will need to be set up depending on the installation.
+
+#### Consul UI
+
+Logging in via the Consul UI requires a redirect URI of the form:
+`http://localhost:8500/ui/oidc/callback` or
+`https://{host:port}/ui/oidc/callback`
+
+The "host:port" must be correct for the Consul server.
+
+#### CLI
+
+If you plan to support authentication via `consul login -type=oidc
+-method=<name>`, a localhost redirect URI must be set (usually this is
+`http://localhost:8550/oidc/callback`). Logins via the CLI may specify a
+different host and/or listening port if needed, and a URI with this host/port
+must match one of the configured redirected URIs. These same "localhost" URIs
+must be added to the provider as well.
+
+### OIDC Login
+
+#### Consul UI
+
+**TODO(sso): update this after john is done**
+
+1. Select the "OIDC" login method.
+2. Enter a role name if necessary.
+3. Press "Sign In" and complete the authentication with the configured provider.
+
+#### CLI
+
+```
+$ consul login -method=oidc -type=oidc -token-sink-file=consul.token
+
+Complete the login via your OIDC provider. Launching browser to:
+
+    https://myco.auth0.com/authorize?redirect_uri=http%3A%2F%2Flocalhost%3A8550%2Foidc%2Fcallback&client_id=r3qXc2bix9eF...
+```
+
+The browser will open to the generated URL to complete the provider's login.
+The URL may be entered manually if the browser cannot be automatically opened.
+
+The callback listener may be customized with the following optional parameters.
+These are typically not required to be set:
+
+The callback listener defaults to listen on `localhost:8550`. If you want to
+customize that use the optional flag
+[`-oidc-callback-listen-addr=<host:port>`](/docs/commands/login#oidc-callback-listen-addr).
+
+## OIDC Configuration Troubleshooting
+
+The amount of configuration required for OIDC is relatively small, but it can
+be tricky to debug why things aren't working. Some tips for setting up OIDC:
+
+- Monitor Consul's log output. Important information about OIDC validation
+  failures will be emitted.
+
+- Ensure Redirect URIs are correct in Consul and on the provider. They need to
+  match exactly. Check: http/https, 127.0.0.1/localhost, port numbers, whether
+  trailing slashes are present.
+
+- [`BoundAudiences`](#boundaudiences) is optional and typically
+  not required. OIDC providers will use the `client_id` as the audience and
+  OIDC validation expects this.
+
+- Check your provider for what scopes are required in order to receive all of
+  the information you need. The scopes "profile" and "groups" often need to be
+  requested, and can be added by setting
+  `[OIDCScopes](#oidcscopes)="profile,groups"` on the auth method.
+
+- If you're seeing claim-related errors in logs, review the provider's docs
+  very carefully to see how they're naming and structuring their claims.
+  Depending on the provider, you may be able to construct a simple `curl`
+  implicit grant request to obtain a JWT that you can inspect. An example of
```

Comment:
> construct a simple `curl` implicit grant request
I'm not sure what one of those is. Can we describe that more plainly or potentially link to an external resource that describes how one might do this or what an 'implicit grant request' is?

Reply:
The joys of forking documentation from another product is that some of the things do need a bit of love after forking as Consul's audience may be differently technical than Vault's audience.

Reply:
I've linked `implicit grant` to clear that up.